### ISSUE TITLE
Bug - Select/Dropdown - menutoggle doesn't receive focus on menu close

### ISSUE THREAD

**Issue by nicolethoen** [patternfly/patternfly-react#8985]
For the newly promoted Select & Dropdown components, we need to ensure that browser focus returns back to the MenuToggle:
1. when an MenuItem has been selected and results in the menu closing 
2. when the 'escape' or 'tab' keyboard is used to close the Menu

This is a regression from our deprecated Dropdown and Select implementations

**Comment by kmcfaul** [comment]
Will open a separate PR for dropdown's fix
